src/Interfaces/AbstractZonotope.jl

```diff
@@ -542,25 +544,38 @@ Reachable Sets of Hybrid Systems Using a Combination of Zonotopes and Polytopes.
 2009.*
 
 The one-dimensional case is not covered by that algorithm; we manually handle
-this case, assuming that there is only one generator.
+this case.
 """
 function constraints_list(Z::AbstractZonotope{N}; check_full_rank::Bool=true) where {N<:AbstractFloat}
-    G = genmat(Z)
-    p = ngens(Z)
     n = dim(Z)
 
-    # use fallback implementation if order < 1 or matrix is not full rank
-    if p < n || (check_full_rank && rank(G) < n)
-        return _constraints_list_fallback(Z)
+    # special handling of the 1D case
+    if n == 1
+        return _constraints_list_1d(Z)
     end
 
-    # special handling of 1D case
-    if n == 1
-        c = center(Z, 1)
-        g = sum(abs.(view(G, 1, :)))
-        constraints = [LinearConstraint([N(1)], c + g),
-                       LinearConstraint([N(-1)], g - c)]
-        return constraints
+    G = genmat(Z)
+    p = ngens(Z)
+
+    # check whether to use the fallback implementation
+    use_fallback = false
+    if p < n
+        # order < 1
+        use_fallback = true
+    elseif check_full_rank
```

Comment:
The precondition for the algorithm in [Lemma 7, 1] is that the `p` generators should be "independent" (which obviously doesn't mean linear independence; what is required is that the denominator in `Ci+` is never zero).

It can happen that `p < n` but the `p` generators are independent.

At this point I think we should remove the `rank` criteria and instead apply `remove_redundant_generators` with an optional flag set to `true` by default (the flag we can could call it `prune`).

The suggestion to remove the rank criterion is because it helps only in a few cases and it's a bit complicated. Let `r = rank(G)`, if i'm not mistaken it should be like this:

- If `p <= n` then `r` is at most `p`. If `r = p` then for sure all the generators are independent. Otherwise we need to prune.
 
- If `p > n` then `r` is at most `n`. We can't conclude anything from rank criteria and always have to apply the prune method.

Reply:
@mforets The test in `test/LazyOperations/ResetMap.jl:65` now fails, so the change seems to be incorrect. One of the sets does not use the vrep fallback now and the implementation computes an `HPolytope` with no constraints. I have not checked whether the computation has a bug or whether this input should use the vrep fallback.

Reply:
This error was just a bug. When `remove_redundant_generators` could remove zero columns but nothing else, this fact was not realized.